azurerm/internal/services/cosmos/resource_arm_cosmosdb_mongo_collection.go

```diff
@@ -264,7 +303,16 @@ func resourceArmCosmosDbMongoCollectionRead(d *schema.ResourceData, meta interfa
 		}
 
 		if props.Indexes != nil {
```

Comment:
we need to ensure these fields are always set, so we can remove the if statement here, since the `flatten` function should handle these being nil

Reply:
updated